Partial specialization error with GCC 6.3

On Ubuntu 17.04 with GCC 6.3 and Boost 1.62 compilation of liblsl fails:
<pre><code>/usr/bin/c++   -Dlsl_EXPORTS -ILSL/liblsl -O3 -DNDEBUG -fPIC -fvisibility=hidden -MD -MT LSL/liblsl/CMakeFiles/lsl.dir/src/legacy/legacy_abi.cpp.o -MF LSL/liblsl/CMakeFiles/lsl.dir/src/legacy/legacy_abi.cpp.o.d -o LSL/liblsl/CMakeFiles/lsl.dir/src/legacy/legacy_abi.cpp.o -c LSL/liblsl/src/legacy/legacy_abi.cpp
In file included from /usr/include/boost/container/detail/flat_tree.hpp:29:0,
                 from /usr/include/boost/container/flat_set.hpp:29,
                 from LSL/liblsl/src/legacy/../send_buffer.h:5,
                 from LSL/liblsl/src/legacy/../tcp_server.h:13,
                 from LSL/liblsl/src/legacy/../stream_outlet_impl.h:4,
                 from LSL/liblsl/src/legacy/legacy_abi.cpp:2:
/usr/include/boost/container/detail/pair.hpp:433:8: error: partial specialization of ‘struct boost::is_enum<std::pair<_T1, _T2> >’ after instantiation of ‘struct boost::is_enum<std::pair<double, bool> >’ [-fpermissive]
 struct is_enum< ::std::pair<T, U> >
        ^~~~~~~~~~~~~~~~~~~~~~~~~~~~
</code></pre>

Clang doesn't complain about it (yet?) and `-fpermissive` lets GCC accept it, so we could add that to the compilation flags until we find a permanent solution.

--- a/LSL/liblsl/src/stream_outlet_impl.h
+++ b/LSL/liblsl/src/stream_outlet_impl.h
@@ -1,6 +1,7 @@
 #ifndef STREAM_OUTLET_IMPL_H
 #define STREAM_OUTLET_IMPL_H
 
+#include <boost/container/flat_set.hpp>
 #include "tcp_server.h"
 #include "send_buffer.h"
 #include "common.h"

--- a/LSL/liblsl/src/tcp_server.cpp
+++ b/LSL/liblsl/src/tcp_server.cpp
@@ -5,6 +5,7 @@
 #include <boost/uuid/uuid_generators.hpp>
 #include <boost/uuid/uuid_io.hpp>
 #include <boost/serialization/split_member.hpp>
+#include <boost/container/flat_set.hpp>
 #include "tcp_server.h"
 #include "socket_utils.h"
 

